### DISCUSSION TITLE
Using AWS Managed elasticSearch 7.4 with Conductor 3.5.1

### DISCUSSION THREAD

**Discussion by shekarpcs** [Netflix/conductor#2789] [Q&A] [answered]
I am receiving following error when I connect to AWS elasticSearch 7.4. Conductor Server is 3.5.1
Is there any setting which I need to make to connect to AWS Managed ElasticSearch

9024 [main] ERROR org.springframework.boot.SpringApplication [] - Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'queueAdminResource' defined in URL [jar:file:/Users/xxxx/conductor-config/conductor-server-3.5.1-boot.jar!/BOOT-INF/lib/conductor-rest-3.5.1.jar!/com/netflix/conductor/rest/controllers/QueueAdminResource.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'defaultEventQueueProcessor' defined in URL [jar:file:/Users/xxxx/conductor-config/conductor-server-3.5.1-boot.jar!/BOOT-INF/lib/conductor-core-3.5.1.jar!/com/netflix/conductor/core/events/queue/DefaultEventQueueProcessor.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'workflowExecutor' defined in URL [jar:file:/Users/xxx/conductor-config/conductor-server-3.5.1-boot.jar!/BOOT-INF/lib/conductor-core-3.5.1.jar!/com/netflix/conductor/core/execution/WorkflowExecutor.class]: Unsatisfied dependency expressed through constructor parameter 5; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'executionDAOFacade' defined in URL [jar:file:/Users/xxx/conductor-config/conductor-server-3.5.1-boot.jar!/BOOT-INF/lib/conductor-core-3.5.1.jar!/com/netflix/conductor/core/dal/ExecutionDAOFacade.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'es6IndexRestDAO': Invocation of init method failed; nested exception is java.net.ConnectException: Connection refused

**Comment by jxu-nflx** [comment]
It looks like it still trying to build using es6, did you change conductor.elasticsearch.version from 6 to 7?

**Comment by shekarpcs** [comment]
I used apache elastic version in local environment. It was working fine.  It is throwing the error when I configured AWS managed elastic search

**Reply by jxu-nflx** [reply]
@shekarpcs when you run it in local, does it intizialze the es7 bean or the es6 bean. As I saw in the exception stack trace you posted above

> nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'es6IndexRestDAO': Invocation of init method failed; nested exception is java.net.ConnectException: Connection refused

It tries to initize a es6 bean.
Can you also post your config file as well?

**Reply by shekarpcs** [reply]
> @shekarpcs when you run it in local, does it intizialze the es7 bean or the es6 bean. As I saw in the exception stack trace you posted above
> 
> > nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'es6IndexRestDAO': Invocation of init method failed; nested exception is java.net.ConnectException: Connection refused
> 
> It tries to initize a es6 bean. Can you also post your config file as well?


I  have tried to change the elastic search version in config file 

conductor.elasticsearch.version=7

I got following error. 


Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
8641 [main] ERROR org.springframework.boot.diagnostics.LoggingFailureAnalysisReporter [] - 

***************************
APPLICATION FAILED TO START
***************************

Description:


An attempt was made to call a method that does not exist. The attempt was made from the following location:

    com.netflix.conductor.es7.dao.index.ElasticSearchRestDAOV7.waitForHealthyCluster(ElasticSearchRestDAOV7.java:324)

The following method did not exist:

    'void org.elasticsearch.client.Request.addParameters(java.util.Map)'

The method's class, org.elasticsearch.client.Request, is available from the following locations:

    jar:file:/Users//conductor-config/conductor-server-3.5.1-boot.jar!/BOOT-INF/lib/elasticsearch-rest-client-6.8.12.jar!/org/elasticsearch/client/Request.class

The class hierarchy was loaded from the following locations:

    org.elasticsearch.client.Request: jar:file:/Users//conductor-config/conductor-server-3.5.1-boot.jar!/BOOT-INF/lib/elasticsearch-rest-client-6.8.12.jar!/

**Reply by shekarpcs** [reply]
Should I updated any jar file or classpath to refer ES7?

**Comment by jxu-nflx** [comment]
Did you follow all the steps [listed](https://github.com/Netflix/conductor/blob/main/es7-persistence/README.md) here?

**Reply by shekarpcs** [reply]
> Did you follow all the steps [listed](https://github.com/Netflix/conductor/blob/main/es7-persistence/README.md) here?
We are not building Conductor server jar locally instead we are dowloading from maven repository.  We tried to use AWS ES6 implementation but it does not have mechanism to pass basic authentication uid/pwd. We are stuck as our AWS ES6 and ES7 needs basic authentication

**Reply by jxu-nflx** [reply]
Hello @shekarpcs the currently release defaults to use ES6, and unfortunately ES6 doesn't have the support to do basic auth. So instead of custom building the conductor to use ES7, another option is to add the basic auth support to ES6. Feel free to submit a PR for it.

**Comment by erikbrgr** [comment, marked as answer]
If anyone comes to this discussion from Google, I wrote an article about running Conductor using Elasticsearch 7. Maybe it'll help.

https://betterprogramming.pub/running-netflix-conductor-3-in-docker-using-elasticsearch-7-and-postgresql-b415988dd74a
